src/Workspaces/Core/Portable/EmbeddedLanguages/RegularExpressions/LanguageServices/RegexEmbeddedHighlighter.cs

```diff
@@ -0,0 +1,159 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.Common;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.LanguageServices;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.VirtualChars;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.EmbeddedLanguages.RegularExpressions.LanguageServices
+{
+    using RegexToken = EmbeddedSyntaxToken<RegexKind>;
+
+    internal class RegexEmbeddedHighlighter : IEmbeddedHighlighter
+    {
+        private readonly RegexEmbeddedLanguage _language;
+
+        public RegexEmbeddedHighlighter(RegexEmbeddedLanguage language)
+        {
+            _language = language;
+        }
+
+        public async Task<ImmutableArray<TextSpan>> GetHighlightsAsync(Document document, int position, CancellationToken cancellationToken)
+        {
+            var option = document.Project.Solution.Workspace.Options.GetOption(RegularExpressionsOptions.HighlightRelatedRegexComponentsUnderCursor, document.Project.Language);
+            if (!option)
+            {
+                return default;
```

Comment:
>default [](start = 23, length = 7)

nit: could be null here and below as well.

Reply:
Can't be. it's an ImmutableArray.